File: segundo/1er cuatri/PDOO/ejercicios_clase/clase25nov/src/clase25nov/Personaje.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package clase25nov;

/**
 *
 * @author alvarogz
 */
public abstract class Personaje extends SerVivo{
    public static String TIPO_MONSTRUO = "MONSTRUO";
    public static String TIPO_JUGADOR = "JUGADOR";
    protected String tipo;

    public Personaje(String nombre, int vida) {
        super(nombre, vida);
    }

    public String getTipo() {
        return tipo;
    }

    public abstract void atacar();
    
    public abstract void defender();
}

File: segundo/1er cuatri/PDOO/ejercicios_clase/clase25nov_ruby/personaje.rb
# frozen_string_literal: true
require_relative 'ser_vivo'

class Personaje < SerVivo
  @@TIPO_MONSTRUO = "MONSTRUO"
  @@TIPO_JUGADOR = "JUGADOR"

  def initialize(nombre, vida)
    super(nombre, vida)
    @tipo
  end

  attr_reader :tipo

  def atacar
    raise "Metodo abstracto"

  end

  def defender
    raise "Metodo abstracto"
  end
end
